### PR TITLE
FCT anchors specification

### PR DESCRIPTION
**Pull request by ilzheev**
This PR adds structs and functions related to FCT anchoring (anchors from customnet into Factom mainnet).
As BTC and ETH anchoring is not compatible (logically) with FCT anchoring (i.e. mainnet is anchored into BTC and ETH, and customnet is anchored into mainnet, but not into BTC and ETH), so I created separate structs and functions for Factom Anchors.

 _Basically it's a copy of default Anchors structs and functions:_

1. AnchorsFactom{}
2. AnchorFactom{}
3. (a *AnchorsFactom) String()
4. (a *AnchorsFactom) UnmarshalJSON()
5. getAnchorsFactom(hash string, height int64) (*AnchorsFactom, error)
6. GetAnchorsFactom(hash string) (*AnchorsFactom, error)
7. GetAnchorsFactomByHeight(height int64) (*AnchorsFactom, error)

Also there is a test for getAnchorsFactom

### CODE REVIEW

File: anchors.go [hunk 1]
```diff
@@ -93,6 +124,14 @@ func (a *Anchors) UnmarshalJSON(data []byte) error {
 	return json.Unmarshal(data, tmp(a))
 }
 
+// UnmarshalJSON is an unmarshaller that handles the variable response from factomd
+func (a *AnchorsFactom) UnmarshalJSON(data []byte) error {
+	type tmp *AnchorsFactom // unmarshal into a new type to prevent infinite loop
+	// json can't unmarshal a bool into a struct, but it can recognize a null pointer
+	data = bytes.Replace(data, []byte("\"factom\":false"), []byte("\"factom\":null"), -1)
```
Comment (WhoSoup) [inline anchors.go:131]: This was a workaround in Anchors.UnmarshalJSON because the Factomd API wrongfully reports it like that and would have required a breaking API change to fix. Since this doesn't have an implemented on the factomd end yet, I see no reason to do it this way and just have the API implementation report a null.

**Review by WhoSoup — COMMENTED** [review]
Is there a reference document for the functionality of this? I don't see a PR for the factomd implementation of this, so it's hard to review the functionality at this point.

**Comment by ilzheev** [comment]
There is no factomd implementation yet, so I agree that we should merge this PR together with factomd PR related to FCT anchors. 

Is anyone willing to develop factomd functionality for this?
@WhoSoup @PaulBernier @PaulSnow

**Comment by WhoSoup** [comment]
> Is anyone willing to develop factomd functionality for this?

Is there a technical design for how the thing is supposed to work (in terms of parsing, config layout, format, etc) available?

**Comment by ilzheev** [comment]
> > Is anyone willing to develop factomd functionality for this?
> 
> Is there a technical design for how the thing is supposed to work (in terms of parsing, config layout, format, etc) available?

It's pretty the same as mainnet BTC anchoring.

- The anchoring software (standalone binary) connects to both custom network and Factom Open Node
- Chain for anchors is created on mainnet, the chainID set into anchoring software config
- Chain for proof-of-anchors (FactomAnchorChainFCT) is created on customnet
- Customnet factomd authority nodes add ChainID of FactomAnchorChain into config and parse anchoring data from there

TL;DR;
Anchoring software (to be developed by De Facto):
- Reads customnet dblocks and anchors them into mainnet
- Writes proof-of-anchors into FactomAnchorChain on customnet

Customet factomd:
- Parses data from FactomAnchorChain and return it via anchors API

Testnet/customnet explorer:
- Parses data from factomd via anchors API

--
Need to develop:
1) specification for proof-of-anchor entries into customnet FactomAnchorChainFCT
2) specification for anchor entries into mainnet
3) factomd API update to support parsing FCT anchors from FactomAnchorChainFCT
